rpc/client.go

```diff
@@ -109,6 +157,115 @@ func (opts *UploadReportOptions) artifactConsumer(ctx context.Context, testChan
 	}
 }
 
+func (opts *UploadReportOptions) validate() error {
+	catcher := grip.NewBasicCatcher()
+
+	catcher.NewWhen(opts.AWSAccessKey == "", "must provide an AWS access key")
+	catcher.NewWhen(opts.AWSSecretKey == "", "must provide an AWS secret key")
+	catcher.NewWhen(opts.DataPipesHost == "", "must provide the Data Pipes service hostname")
+	catcher.NewWhen(opts.DataPipesRegion == "", "must provide the Data Pipes service region")
+
+	return errors.Wrap(catcher.Resolve(), "invalid upload report options")
+}
+
+// getSignedURL calls the Data Pipes API to retrieve a signed URL, where it can PUT the report JSON.
+// Data Pipes docs: https://github.com/10gen/data-pipes.
+func getSignedURL(opts *UploadReportOptions) (string, error) {
+	service := "execute-api"
+	resultType := "cedar-report"
+	if opts.AWSAccessKey == "" || opts.AWSSecretKey == "" || opts.DataPipesHost == "" || opts.DataPipesRegion == "" {
+		return "", errors.New("Getting signed URL failed. AWS access key, AWS secret key, data pipes host and data pipes region required.")
+	}
+
+	// See the Data Pipes documentation for more information.
+	name := uuid.New()
+	url := fmt.Sprintf("%s/v1/results/evergreen/%s/%s/%s/%s", opts.DataPipesHost, opts.Report.TaskID, strconv.Itoa(opts.Report.Execution), resultType, name.String())
+	grip.Debug(message.Fields{
+		"request_url": url,
+	})
+	//bytes.NewBuffer(data)
+	req, err := http.NewRequest(http.MethodPut, url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	AWSCredentials := credentials.NewStaticCredentials(opts.AWSAccessKey, opts.AWSSecretKey, opts.AWSToken)
+	signer := v4.NewSigner(AWSCredentials)
+	_, err = signer.Sign(req, nil, service, opts.DataPipesRegion, time.Now())
+	if err != nil {
+		return "", err
+	}
+
+	response, err := opts.DataPipesHTTPClient.Do(req)
+	if err != nil {
+		return "", errors.Wrap(err, "failed getting signed url")
```

Comment:
Our error message convention is just to state the failed action, you can omit `failed` or `error`. This would be then `getting signed URL`. Since we wrap our errors we found that it is redundant an unnecessary.